libvcs4j-mapping/src/main/java/de/unibremen/informatik/st/libvcs4j/mapping/Tracker.java

```diff
@@ -36,6 +38,102 @@
 	public void add(final Mapping.Result<T> result)
 			throws NullPointerException {
 		Validate.notNull(result);
-		throw new UnsupportedOperationException("Not yet implemented");
+		final Map<Mappable<T>, Lifespan<T>> localMappables = new IdentityHashMap<>();
+		final List<Mappable<T>> mappedTo = result.getTo();
+
+		if (mappables.isEmpty()) {
+			mappedTo.forEach(to ->
+					convertToEntityAndAddToMap(to,
+							result.getOrdinal(),
+							localMappables));
+		} else {
+			mappedTo.stream()
+					.filter(mappable -> mappables.containsKey(
+							result.getPredecessor(mappable).get()))
+					.forEach(to -> {
+						final Mappable<T> predecessor =
+								result.getPredecessor(to).get();
+						final Entity<T> last = mappables.get(predecessor).getLast();
+						final int numChanges = getNumChanges(predecessor,
+								to,
+								last.getNumChanges());
+						final Entity<T> successor =
+								new Entity<>(to, result.getOrdinal(), numChanges);
+						//just to be sure
+						if (last.getOrdinal() < successor.getOrdinal()) {
+							final Lifespan<T> updated =
+									mappables.get(predecessor).add(successor);
+							localMappables.put(to, updated);
+							if (!lifespans.contains(updated)) {
+								lifespans.add(updated);
+							}
+						}
+
+					});
+		}
+
+
+		final List<Mappable<T>> startingLifespans = result.getStartingLifespans();
+		startingLifespans.forEach(mappable -> convertToEntityAndAddToMap(
+				mappable, result.getOrdinal(), localMappables));
+
+		mappables = localMappables;
+	}
+
+	/**
+	 * Converts a given mappable to a {@link Entity} and adds it the given map.
+	 * A {@link Lifespan} is created aswell. This method is used as a utility
+	 * method by {@link #add(Mapping.Result)}.
+	 *
+	 * @param mappable
+	 * 		The mappable which should be converted.
+	 * @param ordinal
+	 * 		The ordinal of the entity to create. Corresponds to the value of
+	 * 		{@link Mapping.Result#getOrdinal()}.
+	 * @param map
+	 * 		The map on which the new lifespan an its corresponding
+	 * 		{@link Lifespan} should be put.
+	 */
+	private void convertToEntityAndAddToMap(final Mappable<T> mappable,
+											final int ordinal,
+											final Map<Mappable<T>, Lifespan<T>> map) {
+		final Entity<T> entity = new Entity<>(mappable, ordinal, 1);
+		final Lifespan<T> startingLifespan = new Lifespan<>(entity);
+		lifespans.add(startingLifespan);
+		map.put(mappable, startingLifespan);
+	}
+
+	/**
+	 * Increments the given last number of changes, if there is a change present.
+	 *
+	 * @param predecessor
+	 * 		The predecessor mappable.
+	 * @param successor
+	 *		The successor mappable.
+	 * @param lastNumChanges
+	 * 		The number of changes, that will be incremented.
+	 * @return
+	 * 		The incremented number of changes.
+	 */
+	private int getNumChanges(final Mappable<T> predecessor,
```

Comment:
The name of this method is misleading. According to the documentation it only increments `lastNumChanges`, if a change is present. It does not compute the number of actual changes.